src/search/search_encoding.h

```diff
@@ -76,6 +77,24 @@ enum class IndexFieldType : uint8_t {
   TAG = 1,
 
   NUMERIC = 2,
+
+  VECTOR = 3,
+};
+
+enum class VectorType : uint8_t {
+  FLOAT32 = 0,
+  FLOAT64 = 1,
```

Comment:
```suggestion
  FLOAT64 = 0,
```

maybe we can remove float32 if it's currently not supported.